### PR TITLE
first commit

### PR DESCRIPTION
**Pull request by hjain-git**
WebRTC with Browserstack automate for review.

### CODE REVIEW

File: src/main/java/WebRTCConf.java [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+      Random random = new Random();
+        int roomId = random.nextInt(500000);
+        roomId += 100000;
+        
+        return roomId;
+   }
+   
+   public static ChromeOptions getFireFoxConfiguration() {
+      
+      FirefoxProfile firefoxProfile = new FirefoxProfile();
```
Comment (samirans89) [inline src/main/java/WebRTCConf.java]: use formatter for indentation. e.g. Shift + Option + F in VS code.

File: src/main/java/WebRTCConf.java [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+   
+   public static int getRoomID() {
+      Random random = new Random();
+        int roomId = random.nextInt(500000);
+        roomId += 100000;
+        
+        return roomId;
+   }
+   
+   public static ChromeOptions getFireFoxConfiguration() {
```
Comment (samirans89) [inline src/main/java/WebRTCConf.java]: check if ChromeOptions has a parent class or can we specify correct options class as return type for each browser?

File: src/main/java/WebRTCConf.java [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+      
+      FirefoxProfile firefoxProfile = new FirefoxProfile();
+        firefoxProfile.setPreference("media.navigator.streams.fake", true);
+        FirefoxOptions firefoxOptions = new FirefoxOptions();
+        firefoxOptions.setProfile(firefoxProfile);
+        
+        ChromeOptions options = new ChromeOptions();
```
Comment (samirans89) [inline src/main/java/WebRTCConf.java]: Does this need to be ChromeOptions or can stay as FirefoxOptions?

File: src/main/java/WebRTCConf.java [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+        
+        return options;
+   }
+
+    public static ChromeOptions getEdgeConfiguration() {
+
+        EdgeOptions edgeOptions = new EdgeOptions();
+
+        ChromeOptions options = new ChromeOptions();
+        options.merge(edgeOptions);
```
Comment (samirans89) [inline src/main/java/WebRTCConf.java]: There is currently nothing in edgeOptions methods explicitly added? Is this needed?

File: src/main/java/WebRTCConf.java [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+        options.setCapability("os", "Windows");
+        options.setCapability("os_version", "10");
+        options.setCapability("build", "WebRTC Dummy Video Call Build - Edge-Safari");
+        options.setCapability("name", "WebRTC Room Creation - Edge Browser");
+        options.setCapability("browserstack.idleTimeout", 300);
+
+        return options;
+    }
+
+    public static ChromeOptions getSafariConfiguration() {
```
Comment (samirans89) [inline src/main/java/WebRTCConf.java]: same question for the return type - can it be SafariOptions or a parent class of Chrome & Safari options?

File: src/main/java/WebRTCConf.java [hunk 6]
```diff
@@ -0,0 +1,244 @@
...
+        options.setCapability("build", "WebRTC Dummy Video Call Build - Android-iOS");
+        options.setCapability("name", "WebRTC Room Creation - Android");
+        options.setCapability("browserstack.idleTimeout", 300);
+        
+        return options;
+    }
+   
+   public static void main(String[] args) {
+	   	//Disabling Selenium messages.
+	   	java.util.logging.Logger.getLogger("org.openqa.selenium").setLevel(Level.OFF);
```
Comment (samirans89) [inline src/main/java/WebRTCConf.java]: Is this a must have?

File: src/main/java/WebRTCConf.java [hunk 7]
```diff
@@ -0,0 +1,244 @@
...
+           int roomId = getRoomID();
+            
+        	//Getting User's Selection
+        	Scanner sc = new Scanner(System.in);
+        	System.out.println("Please Select a Browser Combination to initiate the WebRTC test on BrowserStack : ");
+        	System.out.println("1 : Press 1 to initiate WebRTC test on Google Chrome and Firefox Browser Combination ");
+        	System.out.println("2 : Press 2 to initiate WebRTC test on Edge and Safari Browser Combination ");
+        	System.out.println("3 : Press 3 to initiate WebRTC test on Android and iOS Combination");
+        	int userSelection  = sc.nextInt();
+        	sc.close();
```
Comment (samirans89) [inline src/main/java/WebRTCConf.java]: This line should be in its own finally block. Else we could have an object never closed situation here. leading to memory leak

File: src/main/java/WebRTCConf.java [hunk 8]
```diff
@@ -0,0 +1,244 @@
...
+        	Thread t1 = null;
+        	Thread t2 = null;
```
Comment (samirans89) [inline src/main/java/WebRTCConf.java]: please have more appropriate naming for these thread objects

File: src/main/java/WebRTCConf.java [hunk 9]
```diff
@@ -0,0 +1,244 @@
...
+        	
+        	if(userSelection == 1) {
```
Comment (samirans89) [inline src/main/java/WebRTCConf.java]: use switch case for more than 2 cases.

File: src/main/java/WebRTCConf.java [hunk 10]
```diff
@@ -0,0 +1,244 @@
...
+                t1 = new Thread(new WebRTCTestRunner(edgeConfiguration, String.valueOf(roomId),false,20000, userSelection));
+                t1.start();
+                
+                //Waiting for other user to join the above created WebRTC room.
+                Thread.sleep(15000);
+                
+                // Joining the above created Room on Safari Browser.
+                System.out.println("Joining the Room on Safari Browser");
+                t2 = new Thread(new WebRTCTestRunner(safariConfiguration,String.valueOf(roomId),true,20000, userSelection));
+                t2.start();
```
Comment (samirans89) [inline src/main/java/WebRTCConf.java]: can we have all this code as a common separate function? and call it with different parameter. this way we can reduce the repeated code

File: src/main/java/WebRTCTestRunner.java [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+import org.openqa.selenium.chrome.ChromeOptions;
+import org.openqa.selenium.remote.RemoteWebDriver;
+
+public class WebRTCTestRunner implements Runnable {
+
+	
+	public static final String USERNAME = System.getenv("BROWSERSTACK_USERNAME");
+   public static final String ACCESS_KEY = System.getenv("BROWSERSTACK_ACCESS_KEY");
+
+    public static final String HUB_URL = "https://" + USERNAME + ":" + ACCESS_KEY + "@hub-cloud.browserstack.com/wd/hub";
```
Comment (samirans89) [inline src/main/java/WebRTCTestRunner.java]: refrain from using USERNAME and ACCESS_KEY in the URL. use browserstack capabilities: browserstack.user and browserstack.key instead

File: src/main/java/WebRTCTestRunner.java [hunk 12]
```diff
@@ -0,0 +1,84 @@
...
+            driver = new RemoteWebDriver(new URL(HUB_URL), options);
+            
+            //Creating new WebRTC Room with generated roomID
+            if(!joinExisting) {
+            	
+            	if(userSelection != 3) {
+                	driver.manage().window().maximize();
+                }
+            	
+                driver.get("https://appr.tc/");
```
Comment (samirans89) [inline src/main/java/WebRTCTestRunner.java]: have string variables like these defined at the top of the class as distinct vars

**Review by samirans89 — CHANGES_REQUESTED** [review]

File: src/main/java/WebRTCConf.java (hunk 6 above)
Comment (hjain-git) [reply src/main/java/WebRTCConf.java]: This is added to disable Selenium INFO logs on terminal

image

File: src/main/java/WebRTCConf.java (hunk 8 above)
Comment (hjain-git) [reply src/main/java/WebRTCConf.java]: done

File: src/main/java/WebRTCConf.java (hunk 9 above)
Comment (hjain-git) [reply src/main/java/WebRTCConf.java]: done

File: src/main/java/WebRTCTestRunner.java (hunk 12 above)
Comment (hjain-git) [reply src/main/java/WebRTCTestRunner.java]: done

File: src/main/java/WebRTCConf.java (hunk 1 above)
Comment (hjain-git) [reply src/main/java/WebRTCConf.java]: done

File: src/main/java/WebRTCConf.java (hunk 7 above)
Comment (hjain-git) [reply src/main/java/WebRTCConf.java]: done

File: src/main/java/WebRTCConf.java (hunk 2 above)
Comment (hjain-git) [reply src/main/java/WebRTCConf.java]: ChromeOptions has MutableCapabilities as parent class. Modified the code accordingly.

File: src/main/java/WebRTCConf.java (hunk 3 above)
Comment (hjain-git) [reply src/main/java/WebRTCConf.java]: done

File: src/main/java/WebRTCConf.java (hunk 4 above)
Comment (hjain-git) [reply src/main/java/WebRTCConf.java]: done

File: src/main/java/WebRTCConf.java (hunk 10 above)
Comment (hjain-git) [reply src/main/java/WebRTCConf.java]: done

File: src/main/java/WebRTCTestRunner.java (hunk 11 above)
Comment (hjain-git) [reply src/main/java/WebRTCTestRunner.java]: done

File: src/main/java/WebRTCConf.java (hunk 5 above)
Comment (hjain-git) [reply src/main/java/WebRTCConf.java]: done

**Pull request closed without merging**